server/sonar-server/src/main/java/org/sonar/server/settings/ws/SetAction.java

```diff
@@ -97,6 +108,24 @@ public void handle(Request request, Response response) throws Exception {
     response.noContent();
   }
 
+  private void validate(SetRequest request, Optional<ComponentDto> component) {
+    PropertyDefinition definition = propertyDefinitions.get(request.getKey());
+    if (definition == null) {
+      return;
+    }
+
+    PropertyDefinition.Result result = definition.validate(request.getValue());
+
+    checkRequest(result.isValid(),
+      i18n.message(Locale.ENGLISH, "property.error." + result.getErrorKey(), "Error when validating setting with key '%s' and value '%s'"),
+      request.getKey(), request.getValue());
+
+    checkRequest(component.isPresent() || definition.global(), "Setting '%s' cannot be global", request.getKey());
+    checkRequest(!component.isPresent()
+      || definition.qualifiers().contains(component.get().qualifier()),
+      "Setting '%s' cannot be set on a component with qualifier '%s'", request.getKey(), component.get().qualifier());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Call "component.isPresent()" before accessing the value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS3655)